### DISCUSSION TITLE
"soft_wrap": "editor_width" doesn't work (at least not the way I expect it to)

### DISCUSSION THREAD

**Discussion by rcrooks** [zed-industries/zed#6543] [Feature Requests]
I turned this on, thinking it would wrap lines in the editor, but not actually create line breaks. But it does create breaks.

Does anyone know if this is considered a bug that will be fixed?

**Comment by iamnbutler** [comment]
Hey there, I'm a bit confused if you are seeing something different than me. What are you seeing?

I just tried this right now and this is the experience I have:

https://user-images.githubusercontent.com/1714999/234937628-dbff5d70-e4de-4ef6-96bb-0f788e35795d.mp4

I don't seem to get any line breaks.

**Reply by iamnbutler** [reply]
On stable as well:

https://user-images.githubusercontent.com/1714999/234937974-5ebd32b1-9765-4f79-b485-eaef29030f1b.mp4

Am I misunderstanding the question?

**Reply by rcrooks** [reply]
This is what I'm seeing. Before I add that setting, this is all one line. Opening the file in VS Code confirms it is now multiple lines (the whole file is like that)
<img width="630" alt="2023-04-27_13-56-34" src="https://user-images.githubusercontent.com/619617/234952164-0f3cb8ae-040a-4a6f-8f9d-030636010c07.png">

**Reply by rcrooks** [reply]
Maybe the issue is specific to .html files?

**Reply by iamnbutler** [reply]
Are you sure this is not your formatter? Prettier will do this for example.

Try making a new file, don't save it, enable soft wrap - editor width, make zed smaller and then copy the text without saving. Paste into vscode or another text editor. Are you still seeing that behavior?

To be fair, I don't think I've actually ever edited a raw HTML file in Zed 🤦🏻‍♂️

Let me know how that goes.

cc @JosephTLyons @hovsater

**Reply by hovsater** [reply]
I can't reproduce this either. The following HTML does get reformatted by the language server upon saving the file:

**From this**

```html
<!DOCTYPE html>
<html>
<body>
    <p>Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean ultricies mi vitae est. Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra. Vestibulum erat wisi, condimentum sed, commodo vitae, ornare sit amet, wisi. Aenean fermentum, elit eget tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. Donec non enim in turpis pulvinar facilisis. Ut felis. Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis luctus, metus</p>
</body>
</html>
```

**Into this**

```html
<!DOCTYPE html>
<html>

<body>
    <p>Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Vestibulum tortor
        quam, feugiat vitae, ultricies eget, tempor sit amet, ante. Donec eu libero sit amet quam egestas semper. Aenean
        ultricies mi vitae est. Mauris placerat eleifend leo. Quisque sit amet est et sapien ullamcorper pharetra.
        Vestibulum erat wisi, condimentum sed, commodo vitae, ornare sit amet, wisi. Aenean fermentum, elit eget
        tincidunt condimentum, eros ipsum rutrum orci, sagittis tempus lacus enim ac dui. Donec non enim in turpis
        pulvinar facilisis. Ut felis. Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu
        vulputate magna eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor, facilisis
        luctus, metus</p>
</body>

</html>
````

Adding the following into `settings.json` stops the behaviour as expected:

```json
"HTML": {
  "format_on_save": "off"
}
```

<img width="1195" alt="Screenshot 2023-04-27 at 20 59 21" src="https://user-images.githubusercontent.com/503025/234964994-8490fcc7-ddf6-448b-ad8c-005788d53e4d.png">

**Comment by rcrooks** [comment]
Ok, following your steps, it does not happen. However if I copy from an existing HTML file in VSCode into a new file in Zed, all the lines get broken at column 80. I guess I could get around this by increasing the preferred line length to some astronomical number.

**Comment by rcrooks** [comment]
I tried setting `preferred_line_length` to 0 - didn't help. Setting to 20000 is an effective workaround. Turning of `format_on_save` might be too - I didn't try that. Thanks for the help all!

**Reply by hovsater** [reply]
Glad we could help. 🙂
